src/jdk.jdwp.agent/share/native/libjdwp/debugInit.c

```diff
@@ -181,12 +181,12 @@ DEF_Agent_OnLoad(JavaVM *vm, char *options, void *reserved)
     vmInitialized = JNI_FALSE;
     gdata->vmDead = JNI_FALSE;
 
-    jvmtiCompileTimeMajorVersion  = ( JVMTI_VERSION & JVMTI_VERSION_MASK_MAJOR )
-                                        >> JVMTI_VERSION_SHIFT_MAJOR;
-    jvmtiCompileTimeMinorVersion  = ( JVMTI_VERSION & JVMTI_VERSION_MASK_MINOR )
-                                        >> JVMTI_VERSION_SHIFT_MINOR;
-    jvmtiCompileTimeMicroVersion  = ( JVMTI_VERSION & JVMTI_VERSION_MASK_MICRO )
-                                        >> JVMTI_VERSION_SHIFT_MICRO;
+    jvmtiCompileTimeMajorVersion  = ((int)JVMTI_VERSION & (int)JVMTI_VERSION_MASK_MAJOR)
+                                        >> (int)JVMTI_VERSION_SHIFT_MAJOR;
+    jvmtiCompileTimeMinorVersion  = ((int)JVMTI_VERSION & (int)JVMTI_VERSION_MASK_MINOR)
+                                        >> (int)JVMTI_VERSION_SHIFT_MINOR;
+    jvmtiCompileTimeMicroVersion  = ((int)JVMTI_VERSION & (int)JVMTI_VERSION_MASK_MICRO)
+                                        >> (int)JVMTI_VERSION_SHIFT_MICRO;
```

Comment:
Can you explain the need for this. These types are all naturally jints, so I'm not sure why you would need to cast them here.

Reply:
These are enums (i.e. ints) , GHA on Windows fail because they are 3 different enum types (warning 5287)
For unknown reason (different MSVC version maybe) my local Windows build and our CI passed

Reply:
Similarly, no warnings for `JvmtiExport::decode_version_values()`
https://github.com/openjdk/jdk/blob/e1d81c0946364a266a006481a8fbbac24c7e6c6a/src/hotspot/share/prims/jvmtiExport.cpp#L643-L648
where the version is a `jint`.
And `jint` does seem like the natural type here.